hcl/eval/eval.go

```diff
@@ -57,8 +57,8 @@ func (c *Context) SetFunction(name string, fn function.Function) {
 func (c *Context) SetEnv(environ []string) {
 	env := map[string]cty.Value{}
 	for _, v := range environ {
-		parsed := strings.Split(v, "=")
```

Comment:
Why don't we use `strings.SplitN(v, "=", 1)` instead? [Link](https://pkg.go.dev/strings#SplitN) I think it would be more intuitive

Reply:
yes `SplitN()` does the job but I think I will keep just the `Index()` because we are going make Terramate more GC friendly.
The `SplitN()` does:
```
if n == 0 {
	return nil
}
if sep == "" {
	return explode(s, n)
}
if n < 0 {
	n = Count(s, sep) + 1
}

if n > len(s)+1 {
	n = len(s) + 1
}
a := make([]string, n)
n--
i := 0
for i < n {
	m := Index(s, sep)
	if m < 0 {
		break
	}
	a[i] = s[:m+sepSave]
		s = s[m+len(sep):]
		i++
	}
a[i] = s
return a[:i+1]
```

all those `ifs` are just branch predictors nightmare because `strings.Split()` is a very common function used in many different ways, then the processor is unable to know which branch is more plausible. We also don't need to allocate a new array+slice header.
In our case, just the last bits of this function makes sense, the thing I implemented.